### ISSUE TITLE
Add global option for scratch path location

### ISSUE THREAD

**Issue by jthomasmock** [rstudio/rstudio#15098]
As an extension of #14875 -- we should add a per-user/global option for teams that have their home directory in a slow drive environment all the time (ie in server/Workbench).

**Comment by jthomasmock** [comment]
🚌 Tom

**Comment by gtritchie** [comment]
Using Mac desktop build 2024.11.0-daily+352):

- mkdir ~/foo
- start RStudio, Global Options / General / Advanced: “Default project user data directory” -> ~/foo
- restart RStudio
- create a new Project
- The ~/foo folder contains nothing. Doesn’t appear that the scratch folder is being redirected. I don’t have a per-project folder configured, and don’t have a /etc/rstudio/rsession.conf

Possibly related to handling of `~` as it works if I specify a path outside user directory, e.g. `/tmp/foo`.

**Comment by gtritchie** [comment]
Another scenario not working:

- ensure /tmp/foo exists and is writable
- ensure no “Default project user data directory” is set in RStudio (Global Options / General / Advanced)
- create `/etc/rstudio/rsession.conf`

```
session-project-user-data-dir=/tmp/foo
```

- start RStudio and create a new project
- /tmp/foo is empty, should contain scratch files for that project

**Comment by gtritchie** [comment]
Verified.

The aliased scratch path fix was tested using 2024.11.0-hourly+354 on macOS and I can now specify a scratch path within the home folder. I also checked this on Windows using the same build.

Testing of the global option being specified in rsession.conf via `session-project-user-data-dir=` and the `session-allow-project-user-data-dir-override` setting were validated on Linux (Ubuntu 24) using 2024.11.0-daily+352.
